packages/smithy-http/src/smithy_http/aio/crt.py

```diff
@@ -342,7 +342,7 @@ async def _marshal_request(
         headers_list = []
         if "host" not in request.fields:
             request.fields.set_field(
-                Field(name="host", values=[request.destination.host])
+                Field(name="host", values=[request.destination.netloc])
```

Comment:
We should look into if `netloc` is the right choice. It would follow the format of `{username}:{password}@{host}:{port}`.

I see ruby does `{host}:{port}` ([src](https://github.com/aws/aws-sdk-ruby/blob/version-3/gems/aws-sigv4/lib/aws-sigv4/signer.rb#L636-L643))